test/livebook/runtime/node_pool_test.exs

```diff
@@ -0,0 +1,95 @@
+defmodule Livebook.Runtime.NodePoolTest do
+  use ExUnit.Case, async: true
+
+  alias Livebook.Runtime.NodePool
+
+  # Tests for Livebook.Runtime.NodePool
+  #
+  # Note:
+  # 1. We do not spawn actual nodes as it can be time
+  #    intensive (on low spec machines)
+  #
+  # 2. We wait a certain period of time after we mock
+  #    a node down, so that the buffer period is spent,
+  #    and the node is added to pool.
+
+  describe "start_link" do
+    test "correctly starts a registered GenServer" do
+      start_supervised!({NodePool, name: Foo})
+
+      # Verify Process is running
+      assert Process.whereis(Foo)
+    end
+  end
+
+  describe "get_name/2" do
+    test "creates a new node name if pool is empty" do
+      start_supervised!({NodePool, name: Bar})
+
+      # Assert that we get a result and that it is an atom
+      result = NodePool.get_name(Bar, node())
+      assert result
+      assert is_atom(result)
+    end
+
+    test "adds a new node name to generated names when generated" do
+      start_supervised!({NodePool, name: Baz, buffer_time: 1})
+
+      result = NodePool.get_name(Baz, node())
+
+      # Mock node down
+      send(Baz, {:nodedown, result, {}})
+      Process.sleep(2)
+
+      # Check if node name is added calling get_name/2 again
+      assert NodePool.get_name(Baz, node()) == result
+    end
+
+    test "returns an existing name if pool is not empty" do
```

Comment:
If you look closely this test is exactly the same as `"adds a new node name to generated names when generated"` and `""adds node name to pool"`, so I'd just leave this one (i.e. out of the three this test name is the most descriptive :)).